chore: Add missing code fence languages

nschonni opened the pull request:
<!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [contributing.md](/main/CONTRIBUTING.md#earth_asia-translations) for more information.

Thanks again!
-->

### Why:

Adding the language to the code fences adds syntax highlighting to the samples making them easier to read and can sometimes show syntax errors. The languages can also allow other tools like ESLint/Prettier to format those blocks if they know how to handle them based on the language
<!-- 
- If there's an existing issue for your change, please link to it.
- If there's _not_ an existing issue, please open one first to make it more likely that this update will be accepted: https://github.com/github/docs/issues/new/choose. -->

### What's being changed:

Adding the code languages that are supported by the current syntax highlight library
<!-- Share artifacts of the changes, be they code snippets, GIFs or screenshots; whatever shares the most context. -->

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

nschonni committed:
chore: Add missing code fence languages

janiceilene commented:
👋 @nschonni Thanks for opening a PR! Is this still in progress? When you're ready for review, can you fill out the [pull request template](https://raw.githubusercontent.com/github/docs/main/.github/PULL_REQUEST_TEMPLATE.md) and let me know? Then I'll get it triaged for you!

nschonni commented:
@janiceilene I've checked the checkboxes now

janiceilene commented:
@nschonni Could you fill out the **Why?** and **What's being changed:** sections to help ensure that this is triaged to the right team(s) for review, thanks!

nschonni commented:
@janiceilene done

chiedo commented:
@janiceilene this should be reviewed by someone on the content team! Looks good to me! But with it being a bunch of content changes, I'll defer.

janiceilene commented:
Thanks @chiedo! I'll triage it for a writer review.

rachmari commented on content/actions/creating-actions/setting-exit-codes-for-actions.md:
````diff
@@ -40,7 +40,7 @@ For more information, see "[Creating a JavaScript action](/articles/creating-a-j
 If you are creating a Docker container action, you can set a failure exit code in your `entrypoint.sh` script. For example:
 
-```
+```shell
````
This code fence actually removes `<condition>` in the rendered output. You'll need to add `{% raw %}` and `{% endraw %}` around the code fence. 

https://docs-772--code-fence-languages.herokuapp.com/en/free-pro-team@latest/actions/creating-actions/setting-exit-codes-for-actions#setting-a-failure-exit-code-in-a-docker-container-action

rachmari commented on content/admin/enterprise-management/evacuating-a-cluster-node.md:
````diff
@@ -14,40 +14,40 @@ If you're taking a node offline that has any data services (like git, pages, or
     ```
     For {% data variables.product.prodname_pages %}
-    ```
+    ```shell
````
This code fence doesn't render `<uuid>`. I think we'll need the raw tags around this code fence too. There are several more examples in this file that will need the same treatment.

nschonni replied:
Interesting, I see that to on the preview site now. Is this actually a bug with the rendering library that I should report upstream? It seems odd that it is altering the content of code blocks as `<code class="hljs language-shell">echo "select count(*) from pages_replicas where host = 'pages-server-<uuid>'" | ghe-dbconsole -y</uuid></code>`

rachmari reviewed (commented):
@nschonni I've reviewed the first 13 files in the PR and there are a couple of instances where the code fence removes some information, particularly information in brackets. Can take a look at those instances and any other instances that might be affected and make the appropriate fix? Then I can take a look at this again. 🙏 💚

nschonni commented:
Go away stale bot

chiedo commented:
Coming soon! https://github.com/github/docs/pull/1386

hubwriter commented on content/actions/creating-actions/dockerfile-support-for-github-actions.md:
````diff
@@ -47,20 +47,21 @@ The Docker `ENTRYPOINT` instruction has a _shell_ form and _exec_ form. The Dock
 ##### Example *Dockerfile*
-```	
+
+```	dockerfile
````
```suggestion
```dockerfile
```

hubwriter committed:
Update content/actions/creating-actions/dockerfile-support-for-github-actions.md

hubwriter committed:
Add raw & endraw markers around shell content

See review comment by @rachmari

hubwriter committed:
Add raw & endraw markers around shell content

See review comment by @rachmari

hubwriter committed:
Merge remote-tracking branch 'docs/main' into code-fence-languages

hubwriter reviewed (approved):
As this has been sitting here for a while, I've gone ahead and added the `raw` and `endraw` markers requested by @rachmari.

@rachmari - could you take another look at this? It would be good to get this merged.

@nschonni - Thanks for raising this PR. 👍

hubwriter commented:
A little investigation needed here before we can merge this. 

The `{% raw %} ... {% endraw %}` markers appear not to be working on the staging deployment (e.g. see step 2 [here](https://docs-772--code-fence-languages.herokuapp.com/en/enterprise-server@2.22/admin/enterprise-management/evacuating-a-cluster-node)), although it's working fine on a local deployment.

hubwriter commented:
On investigating this, it seems that this is in fact a rendering issue with our staging deployment on heroku.

See step 2 of this page, where "<uuid>" is rendered correctly in the shell code, without using `{% raw %} ... {% endraw %}`:
https://docs.github.com/en/enterprise-server@2.21/admin/enterprise-management/evacuating-a-cluster-node

Compare this to the staged deployment of this page for the current PR: 
https://docs-772--code-fence-languages.herokuapp.com/en/enterprise-server@2.22/admin/enterprise-management/evacuating-a-cluster-node

rachmari commented:
@hubwriter I'd say that we could just remove the language from the code examples that aren't rendering properly and we can file an issue for any that we can look into the cause for later. I'm not sure when we'll have time to investigate the cause and file an issue with the upstream library, but at least we can ship the updates that are working for us now.

hubwriter commented on content/actions/creating-actions/setting-exit-codes-for-actions.md:
````diff
@@ -41,11 +41,13 @@ For more information, see "[Creating a JavaScript action](/articles/creating-a-j
 
-```
+{% raw %}
+```shell
````
```suggestion
```
```

hubwriter committed:
Remove language from code fences

to avoid the problem of replaceable text indicates like
<this> not showing up in the output page.

hubwriter committed:
Merge remote-tracking branch 'docs/main' into code-fence-languages

hubwriter commented:
I've removed the language from the opening code fence where the code wasn't rendering properly on the output page.

hubwriter commented:
@nschonni - Thanks again for working on this. Apologies it's taken us so long to get it merged. 
It's going in now! 😃
